convert_fendl.py

```diff
@@ -41,90 +42,308 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
-           release_details[args.release]['uncompressed_file_size'])
+WARNING: This script will download {} MB of data.
+Extracting and processing the data requires {} MB of additional free disk space.
+""".format(compressed_file_size, uncompressed_file_size)
 
 # ==============================================================================
 # DOWNLOAD FILES FROM IAEA SITE
 
 if args.download:
     print(download_warning)
-    for f in release_details[args.release]['files']:
-        download(urljoin(release_details[args.release]['base_url'], f),
-                 as_browser=True, context=ssl._create_unverified_context())
+
+    for particle in args.particles:
+        if args.release == '2.1':
+            # Older releases have ace files in individual zip files. Create a 
+            # a directory to hold them.
+            particle_download_path = download_path / particle
+            particle_download_path.mkdir(parents = True, exist_ok=True) 
+            os.chdir(particle_download_path)
+
+        particle_details = release_details[args.release][particle]
+        for f in particle_details['files']:
+            download(urljoin(particle_details['base_url'], f),
+                    as_browser=True, context=ssl._create_unverified_context())
+    
+    os.chdir(cwd)
 
 # ==============================================================================
 # EXTRACT FILES FROM ZIP
 if args.extract:
-    for f in release_details[args.release]['files']:
-        # Extract files, the fendl release was compressed using type 9 zip format
-        # unfortunatly which is incompatible with the standard python zipfile library
-        # therefore the following system command is used
+    for particle in args.particles:
+        if args.release == '2.1':
+            os.chdir(download_path / particle)
 
-        subprocess.call(['unzip', '-o', f, '-d', ace_files_dir])
+        particle_details = release_details[args.release][particle]
+        if particle_details['file_type'] == "ace":
+            extraction_dir = ace_files_dir
+        elif particle_details['file_type'] == "endf":
+            extraction_dir = endf_files_dir
 
-# ==============================================================================
-# GENERATE HDF5 LIBRARY -- NEUTRON FILES
+        for f in particle_details['files']:
+            # Extract files, the fendl release was compressed using type 9 zip format
+            # unfortunatly which is incompatible with the standard python zipfile library
+            # therefore the following system command is used
+            subprocess.call(['unzip', '-o', f, '-d', extraction_dir])
+            if args.cleanup and Path(f).exists():
+                Path(f).unlink()
+        
+    os.chdir(cwd)
+    
+    if args.release == '2.1' and  args.cleanup and download_path.exists():
+        rmtree(download_path)
+
+    # Seperate photon ENDF file for 2.1 release
+    if args.release == '2.1' and 'photon' in args.particles:
```

Comment:
oh wow FENDL 2.1 really bundled everything together like that, wonder if this is worth putting this logic into a function in case we come across this combine ENDF file approach again and keep the flow of the code easier to follow

Reply:
I could sort something out and make a PR to the main OpenMC repo. It would probably fit well in the `data.endf.py` module.

Reply:
Sounds good to me, what do you think @paulromano

Reply:
Just a note here - In theory, we could also use the `get_evaluations` function from the endf module and avoid splitting the file entirely.  Personally I prefer splitting it down so the processing is the same as the other convert scripts.

Reply:
What about always using `get_evaluations`, even when we don't expect there to be more than one evaluation in the file? Seems like that would handle both cases.

Reply:
Less code in the convert_fendl scripts sounds great to me